### PR TITLE
fix(sentry): handle invalid authorization code

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by raisedadead — CHANGES_REQUESTED** [review]
Umm, okay. This fixes a symptom. We want to "handle" the error. The diff between handled and unhandled is exactly as it seems in the name.

The first one is when you know as the developer, that error is something to be expected and conveyed to the user. Example, "Hey! you tried this invalid thing(form input, etc), that's OK, here is your error message in plain english".

The second is something that you do not know as as a developer, as in a use case that you did not anticipate and your code "throws" (or crashes, dies in an unexpected way).


Right now, the logic is unhandled and hence goes to Sentry where we get so see all these unhandled errors and try and fix their cause.

What we want is to convert it into handled error, so we capture it, skip sentry because it need not be seen by us and pass it on the user.

Does this make sense?

With that I think this file is the incorrect location. The most likely are for the fix would be:

https://github.com/freeCodeCamp/freeCodeCamp/blob/4d0cd9e444dcac5c22a1620d8e1c91f403ba9e52/api-server/src/server/component-passport.js#L117-L119

but @ojeytonwilliams has more expertise on this. 

Goodluck!

**Comment by Sembauke** [comment]
Yup that makes sense. Thanks.

**Comment by Sembauke** [comment]
This is not worth my time
